### ISSUE TITLE
Lazy default factory creation

### ISSUE THREAD

**Issue by lowecg** [nilern/Eximia#2]
Your comments suggest that factory creation is expensive:

> Note that XMLInputFactory instances can be very expensive to create but are reusable. So call this as rarely as
  possible, probably only in the application startup phase.

I'm using Woodstox with Eximia and it looks like the default input factory is created eagerly and is unnecessary in my case:

```clojure
(def ^:private ^XMLInputFactory default-input-factory (input-factory {:coalescing true}))
```

I'm using Eximia in AWS Lambda, natively compiled with GraalVM**. The XML libraries can touch *a lot* of classes under the hood. This will cause GraalVM to pull those classes into the native build, so anything that can be done to avoid touching/instantiating unnecessary code will yield tangible improvements in deployment size and startup time (especially from cold starts).

** see https://github.com/FieryCod/holy-lambda

**Comment by nilern** [comment]
I see that lazy default factories definitely make sense.

Although I can't promise it will reduce the image size since Woodstox comes with some config file that makes it the default factory and the factory options might not affect which classes are loaded as much as which methods of the factory are actually used.

**Comment by nilern** [comment]
Lazy factories implemented in [0.1.3](https://github.com/nilern/Eximia/commit/d57c6268eb3ee4a5c3f5baad7336a02693662ae0). It is a bit more efficient if you don't use the default factory but I don't know if this will reduce image size.

If this does not improve image size there might be some other implementation of lazy initialization that does?

**Comment by lowecg** [comment]
thank you for this
